SPECS/netdata.spec

```diff
@@ -1,391 +1,464 @@
+%package freeipmi
+Summary:        FreeIPMI plugin for netdata
+Requires:       %{name}%{?_isa} = %{version}-%{release}
+License:        GPLv3
 
-# XCP-ng: configuration file for netdata-ui
-install -m 644 -p system/netdata.conf "${RPM_BUILD_ROOT}%{_sysconfdir}/%{name}/netdata.conf.ui"
-# XCP-ng: configuration file for netdata-headless
-install -m 644 -p %{SOURCE3} "${RPM_BUILD_ROOT}%{_sysconfdir}/%{name}/netdata.conf.headless"
+%description freeipmi
+freeipmi plugin for netdata
 
-# ###########################################################
-# logrotate settings
-install -m 755 -d "${RPM_BUILD_ROOT}%{_sysconfdir}/logrotate.d"
-install -m 644 -p system/netdata.logrotate "${RPM_BUILD_ROOT}%{_sysconfdir}/logrotate.d/%{name}"
+# XCP-ng: netdata-ui package to enable web UI and open firewall
+%package ui
+BuildArch: noarch
+Summary:   Ready to use netdata for XCP-ng - with web UI enabled
+Requires:  netdata
+# let this package's POST run after that from netdata
+# to avoid a race over the /etc/netdata/netdata.conf symlink
+# (and also we need to restart the netdata service)
+Requires(post): netdata
+# Same for POSTUN
+Requires(postun): netdata
 
-# ###########################################################
-# Install freeipmi
-install -m 4750 -p freeipmi.plugin "${RPM_BUILD_ROOT}%{_libexecdir}/%{name}/plugins.d/freeipmi.plugin"
+%description ui
+Netdata, ready to use on XCP-ng, with web UI enabled.
 
-# ###########################################################
-# Install apps.plugin
-install -m 4750 -p apps.plugin "${RPM_BUILD_ROOT}%{_libexecdir}/%{name}/plugins.d/apps.plugin"
+Installing this package will install netdata with a default
+configuration suitable for XCP-ng.
 
-# ###########################################################
-# Install perf.plugin
-install -m 4750 -p perf.plugin "${RPM_BUILD_ROOT}%{_libexecdir}/%{name}/plugins.d/perf.plugin"
+Warning: this will also open the firewall port 19999 to make
+the readonly web UI available immediately.
 
-# ###########################################################
-# Install cups.plugin
-# XCP-ng: NO, we don't build it
+%post ui
+if [ $1 == 1 ]; then
+    # initial installation
+    if [ -L /etc/netdata/netdata.conf ]; then
+        rm -f /etc/netdata/netdata.conf
+    fi
+    if [ ! -e /etc/netdata/netdata.conf ]; then
+        ln -s netdata.conf.ui /etc/netdata/netdata.conf
+    fi
+    # Restart netdata service
+    /usr/bin/systemctl restart netdata.service
+fi
 
-# ###########################################################
-# Install slabinfo.plugin
-install -m 4750 -p slabinfo.plugin "${RPM_BUILD_ROOT}%{_libexecdir}/%{name}/plugins.d/slabinfo.plugin"
+%postun ui
+if [ $1 == 0 ]; then
+    # uninstallation
+    if [ -L /etc/netdata/netdata.conf ]; then
+        rm -f /etc/netdata/netdata.conf
+    fi
+    if [ ! -e /etc/netdata/netdata.conf ]; then
+        ln -s netdata.conf.headless /etc/netdata/netdata.conf
+    fi
+    # TODO: close firewall port
```

Comment:
Another TODO that we didn't spot in the previous pass.

Reply:
Closing the firewall port is not done yet so it's still a TODO ;)

Reply:
Indeed. I don't know if it's a real TODO though, because a package uninstall + reboot is enough to close it again... Are we going to understand it when we read it in the future?

Reply:
Comment updated